src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -917,6 +917,62 @@ public EventAccessorInfo(MethodInfo interfaceAddMethod, MethodInfo interfaceRemo
                 {
                     InterfaceAddMethod = interfaceAddMethod;
                     InterfaceRemoveMethod = interfaceRemoveMethod;
+
+            private sealed class MethodInfoEqualityComparer : EqualityComparer<MethodInfo>
+            {
+                public static readonly MethodInfoEqualityComparer Instance = new MethodInfoEqualityComparer();
+
+                private MethodInfoEqualityComparer() { }
+
+                public override bool Equals(MethodInfo left, MethodInfo right)
+                {
+                    if (ReferenceEquals(left, right))
+                        return true;
+
+                    if (left == null)
```

Comment:
We are happy if we could use `MetadataToken`... as of ECMA 335, we should check static or instance as well as generic parameters. In addition, we should check their calling conventions. I'll fix it.